cinema/permissions.py

```diff
@@ -0,0 +1,26 @@
+from django.http import Http404
+from rest_framework.permissions import BasePermission
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+
+    def has_permission(self, request, view):
+        return bool(
+            (
+                view.action in ["retrieve", "list"]
```

Comment:
Reformat this line to check request instead of view
```suggestion
                request.method in SAFE_METHODS
```

Reply:
not fixed